### PR TITLE
Financial Connections: added support for skipping success pane

### PR DESCRIPTION
**Pull request by kgaidis-stripe**
## Summary

This PR:
1. Adds "custom public key" and "customer secret key" text fields to the playground (this is so we can test custom merchants)
2. adds support for skip_success_pane

Links:
- https://docs.google.com/document/d/1yTpXDZYQEPms2oRvcjtUdYl2ycsF0XysRQJZq8brmro/edit#heading=h.5b205kqez27g
- https://stripe.slack.com/archives/C01DNBVURH9/p1674066056911119
- https://git.corp.stripe.com/stripe-internal/pay-server/pull/565204

## Testing

Test mode video below. Also tested on live mode.

https://user-images.githubusercontent.com/105514761/215833313-4d0f9975-9321-45d5-b5a3-5e3872cbcad9.mov

### CODE REVIEW

File: Example/FinancialConnections Example/FinancialConnections Example/Playground/PlaygroundMainView.swift [hunk 1]
```diff
@@ -16,50 +16,65 @@ struct PlaygroundMainView: View {
...
             VStack {
                 Form {
-                    VStack(alignment: .leading, spacing: 4) {
-                        Text("What do you want to use Financial Connections for?")
-                            .font(.headline)
-                        Picker("Flow?", selection: $viewModel.flow) {
-                            ForEach(PlaygroundMainViewModel.Flow.allCases) {
-                                Text($0.rawValue.capitalized)
-                                    .tag($0)
+                    Section {
```
Comment (kgaidis-stripe) [inline Example/FinancialConnections Example/FinancialConnections Example/Playground/PlaygroundMainView.swift:19]: all of this code does two things:
1. Adds public key and sercret key fields to playground app
2. Automatically shows/hides some of the fields depending on what is selected (for example, if app to app is enabled, one can't toggle test mode)

File: StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/Models/FinancialConnectionsSessionManifest.swift [hunk 2]
```diff
@@ -76,4 +76,5 @@ struct FinancialConnectionsSessionManifest: Decodable {
     let features: [String: Bool]?
     let experimentAssignments: [String: String]?
     let assignmentEventId: String?
+    let skipSuccessPane: Bool?
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/Models/FinancialConnectionsSessionManifest.swift:79]: backend double-check:
- https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/bank_connections/api/resource/link_account_session_manifest_resource.rb#L85

**Comment by kgaidis-stripe** [comment]
@vardges-stripe I'm publishing this as ready for review, but we will only merge this later after bug-bash

File: Example/FinancialConnections Example/FinancialConnections Example/Playground/PlaygroundMainViewModel.swift [hunk 3]
```diff
@@ -65,6 +65,17 @@ final class PlaygroundMainViewModel: ObservableObject {
...
     }
 
+    @Published var customPublicKey: String = PlaygroundUserDefaults.customPublicKey {
+        didSet {
+            PlaygroundUserDefaults.customPublicKey = customPublicKey
+        }
+    }
+    @Published var customSecretKey: String = PlaygroundUserDefaults.customSecretKey {
+        didSet {
+            PlaygroundUserDefaults.customSecretKey = customSecretKey
```
Comment (vardges-stripe) [inline Example/FinancialConnections Example/FinancialConnections Example/Playground/PlaygroundMainViewModel.swift:75]: feels weird to pass secret key from client. Do we really need this ability to sent whatever key we'd like?

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/NativeFlowController.swift [hunk 4]
```diff
@@ -105,6 +105,19 @@ extension NativeFlowController {
         navigationController.setViewControllers(viewControllers, animated: animated)
     }
 
+    private func pushPane(_ pane: FinancialConnectionsSessionManifest.NextPane, animated: Bool) {
+        if pane == .success && dataManager.manifest.skipSuccessPane == true {
+            closeAuthFlow(showConfirmationAlert: false, error: nil)
```
Comment (vardges-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/NativeFlowController.swift:110]: simple

**Review by vardges-stripe — APPROVED** [review]

File: Example/FinancialConnections Example/FinancialConnections Example/Playground/PlaygroundMainViewModel.swift (hunk 3 above)
Comment (kgaidis-stripe) [reply Example/FinancialConnections Example/FinancialConnections Example/Playground/PlaygroundMainViewModel.swift:75]: @vardges-stripe 

not sure how there is another way for someone to easily (key word) test any set of keys

[dev app has the same function](https://stripejsdevapp.corp.stripe.com/connections-auth-flow)

Comment (vardges-stripe) [reply Example/FinancialConnections Example/FinancialConnections Example/Playground/PlaygroundMainViewModel.swift:75]: Yeah, but Dev app is internal. Saving the security keys in keychain feels ikky, but I guess it's only for internal devices.

**Review by vardges-stripe — DISMISSED** [review]

**Review by vardges-stripe — APPROVED** [review]

**Review by vardges-stripe — APPROVED** [review]

**Pull request merged**